pages/client-libraries/java.mdx

```diff
@@ -4,105 +4,687 @@ description: Peruse our guide on implementing Java in Memgraph. Comprehensive do
+        session.writeTransaction(tx -> {
+            tx.run("CREATE (n:Technology {name:'Memgraph'})");
+            return null;
+        });
+    }
+}
 ```
 
-<Callout type="info">
-Memgraph created [Bolt Java Driver](https://github.com/memgraph/bolt-java-driver) which can be used to connect to a running Memgraph instance. We still recommend you use the above mentioned driver.
-</Callout>
\ No newline at end of file
+The session object takes a SessionConfig object as its argument. The SessionConfig object is used to specify the database name, and other session configuration options. Session API supports writeTransaction and readTransaction methods.
+The writeTransaction method of a Neo4j Session object takes a TransactionWork functional interface as its argument. The TransactionWork interface has a single method execute, which takes a Transaction object as its argument and returns a value of type T, void in this case. 
+
+#### Run a read query 
+
+Similar to the write query, there is a `readTranasction()` method that can be used to run read queries. The difference is that read will actually return the results of the query.
+
+The following query will read data from the database: 
+
+```java
+public void readNode() {
+    try (Session session = driver.session(SessionConfig.forDatabase("memgraph"))) {
+        List<Record> result = session.readTransaction(tx -> {
+            Result queryResult = tx.run("MATCH (n:Technology{name: 'Memgraph'}) RETURN n");
+            List<Record> records = queryResult.list();
+            return records;
+        });
+        System.out.println(result);
+
+        // Print each node as map
+        for (Record record : result) {
+            System.out.println(record.get("n").asMap());
+        }
+    }
+}
+```
+
+In this query, each `Record` contains a `Node` accessible by the `asMap()` method.
+
+#### Run a query with parameters 
+
+If you want to pass a property map to the query, you can do it like this:
+
+```
+public void createNodeWithProperties() {
+    try (Session session = driver.session(SessionConfig.forDatabase("memgraph"))) {
+        Map< String, Object > properties = new HashMap<>();
+        properties.put("name", "Memgraph");
+        properties.put("type", "graph database");
+
+        session.writeTransaction(tx -> {
+            tx.run("CREATE (n:Technology {name: $name, type: $type})", properties);
+            return null;
+        });
+    }
+}
+```
+
+Using this approach, the queries will not contain hard-coded values, they can be more dynamic.
+
+### Process the results
+
+To read results and serve them back to the Java application, they need need to be handled. Java is a statically typed language and proper types need to be handled. 
+Depending on the type of request made, you can receive different results. Let's go over a few basic examples of how to handle different types and access properties of the returned results. 
```

Comment:
```suggestion
In order to serve the read results back to the Java application, their types need to be handled properly because Java is a statically typed language. 
Depending on the type of request made, you can receive different results. Let's go over a few basic examples of how to handle different types and access properties of the returned results.
```